tests/core/pyspec/eth2spec/test/capella/block_processing/test_process_withdrawals.py

```diff
@@ -242,3 +450,395 @@ def test_fail_many_dequeued_incorrectly(spec, state):
+def test_random_partial_withdrawals_5(spec, state):
+    yield from run_random_partial_withdrawals_test(spec, state, random.Random(5))
+
+
+# Tests with multiple blocks
+@with_capella_and_later
+@spec_state_test
+def test_success_two_payloads(spec, state):
+    fully_withdrawable_indices, partial_withdrawals_indices = prepare_expected_withdrawals(
+        spec, state, num_partial_withdrawals=spec.MAX_WITHDRAWALS_PER_PAYLOAD * 4,
+        num_full_withdrawals=spec.MAX_WITHDRAWALS_PER_PAYLOAD * 4)
+
+    next_slot(spec, state)
+    next_withdrawal_index = state.next_withdrawal_index
+    execution_payload = build_empty_execution_payload(spec, state)
+    expected_withdrawals = yield from run_withdrawals_processing(spec, state, execution_payload)
+    verify_post_state(state, spec, expected_withdrawals, fully_withdrawable_indices, partial_withdrawals_indices)
+    withdrawn_indices = [withdrawal.validator_index for withdrawal in expected_withdrawals]
+    fully_withdrawable_indices = list(set(fully_withdrawable_indices).difference(set(withdrawn_indices)))
+    partial_withdrawals_indices = list(set(partial_withdrawals_indices).difference(set(withdrawn_indices)))
+    assert state.next_withdrawal_index == next_withdrawal_index + spec.MAX_WITHDRAWALS_PER_PAYLOAD
+
+    execution_payload = build_empty_execution_payload(spec, state)
+    expected_withdrawals = yield from run_withdrawals_processing(spec, state, execution_payload)
+    verify_post_state(state, spec, expected_withdrawals, fully_withdrawable_indices, partial_withdrawals_indices)
+    assert state.next_withdrawal_index == next_withdrawal_index + spec.MAX_WITHDRAWALS_PER_PAYLOAD * 2
+
+
+@with_capella_and_later
+@spec_state_test
+def test_fail_second_payload_isnt_compatible(spec, state):
+    fully_withdrawable_indices, partial_withdrawals_indices = prepare_expected_withdrawals(
+        spec, state, num_partial_withdrawals=spec.MAX_WITHDRAWALS_PER_PAYLOAD * 4,
+        num_full_withdrawals=spec.MAX_WITHDRAWALS_PER_PAYLOAD * 4)
+
+    next_slot(spec, state)
+    next_withdrawal_index = state.next_withdrawal_index
+    execution_payload = build_empty_execution_payload(spec, state)
+    expected_withdrawals = yield from run_withdrawals_processing(spec, state, execution_payload)
+    verify_post_state(state, spec, expected_withdrawals, fully_withdrawable_indices, partial_withdrawals_indices)
+    withdrawn_indices = [withdrawal.validator_index for withdrawal in expected_withdrawals]
+    fully_withdrawable_indices = list(set(fully_withdrawable_indices).difference(set(withdrawn_indices)))
+    partial_withdrawals_indices = list(set(partial_withdrawals_indices).difference(set(withdrawn_indices)))
+    assert state.next_withdrawal_index == next_withdrawal_index + spec.MAX_WITHDRAWALS_PER_PAYLOAD
+
+    execution_payload = build_empty_execution_payload(spec, state)
+    state.next_withdrawal_index += 1
+    yield from run_withdrawals_processing(spec, state, execution_payload, valid=False)
```

Comment:
Thanks for adding more test cases 👍👍

However, our [current test format](https://github.com/ethereum/consensus-specs/tree/dev/tests/formats/operations) is processing one operation per test case. We can either (1) split `test_success_two_payloads` and `test_fail_second_payload_isnt_compatible` into 4 test cases or (2) I can rewrite them into sanity/block tests.

I think (2) makes more sense here. What do you think?

Reply:
Oh I didn't know that. You can just remove them if you want the real condition is tested in the additional checks added in the post state checker so these tests are just a composition of two tests.

Reply:
Moved them in https://github.com/ethereum/consensus-specs/pull/3068/commits/3fc1ebce06ef5a51a32bb6640507d596358dfc14